uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/WorkPattern.php

```diff
@@ -253,31 +254,94 @@ private static function getMaxWeight() {
    * @return array An array containing the values
    */
   public static function getValuesArray($id) {
-    try {
-      $params = [
-        'id' => $id,
-        'api.WorkWeek.get' => [
-          'pattern_id' => '$value.id',
-          'api.WorkDay.get' => [
-            'week_id' => '$value.id'
-          ]
-        ]
-      ];
-      $result = civicrm_api3('WorkPattern', 'getsingle', $params);
+    $values = [];
+
+    if(empty($id)) {
+      return $values;
+    }
+
+    $workPatternTable = WorkPattern::getTableName();
+    $workWeekTable = WorkWeek::getTableName();
+    $workDayTable = WorkDay::getTableName();
 
-      $workWeeks = $result['api.WorkWeek.get']['values'];
-      foreach($workWeeks as $i => $week) {
-        $workWeeks[$i]['days'] = $week['api.WorkDay.get']['values'];
-        unset($workWeeks[$i]['api.WorkDay.get']);
+    $query = "
+      SELECT
+        wp.id,
+        wp.label,
+        wp.description,
+        wp.is_default,
+        wp.is_active,
+        wp.weight,
+        ww.id as week_id,
+        ww.number,
+        ww.pattern_id,
+        wd.id as work_day_id,
+        wd.day_of_the_week,
+        wd.type,
+        wd.time_from,
+        wd.time_to,
+        wd.break,
+        wd.leave_days,
+        wd.number_of_hours,
+        wd.week_id
+      FROM {$workPatternTable} wp
+      LEFT JOIN {$workWeekTable} ww ON ww.pattern_id = wp.id
+      LEFT JOIN {$workDayTable} wd ON wd.week_id = ww.id
```

Comment:
Because, even though we do some validation on the form level to prevent the user of creating a WorkPattern without at least one Week and WorkDays, there's nothing on the database level to prevent it, which means we can have work patterns without the other child entities.